--- a/spec/cucumber/core/test/filters/locations_filter_spec.rb
+++ b/spec/cucumber/core/test/filters/locations_filter_spec.rb
@@ -376,7 +376,7 @@
       max_duration_ms = max_duration_ms * 2.5 if defined?(JRUBY_VERSION)
       it "filters #{num_features * num_scenarios_per_feature} test cases within #{max_duration_ms}ms" do
         filter = Test::LocationsFilter.new(locations)
-        Timeout.timeout(max_duration_ms / 1000.0) do
+        Timeout.timeout(max_duration_ms / 10.0) do
           compile docs, receiver, [filter]
         end
         expect(receiver.test_cases.length).to eq num_features * num_scenarios_per_feature
